### PR TITLE
[Enhancement] Support describe files()

### PR DESCRIPTION
**Pull request by wyb**
## Why I'm doing:
before this pr, user can get files schema by `create table tbl as select * from files()`, and `desc tbl`

## What I'm doing:
support `desc files()` to get files schema

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/DescribeStmt.java [hunk 1]
```diff
@@ -86,6 +97,14 @@ public DescribeStmt(TableName dbTableName, boolean isAllTables, NodePosition pos
         this.isAllTables = isAllTables;
     }
 
+    public DescribeStmt(Map<String, String> tableFunctionProperties, NodePosition pos) {
+        super(pos);
+        this.dbTableName = null;
+        this.totalRows = new LinkedList<>();
+        this.isTableFunctionTable = true;
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/DescribeStmt.java:104]: We will design more `DESC xxx` in the future, like `DESC credential`, `DESC file format`, besides table, mv, and files table function.
So, it's recommended to use desc_object_type ti distinguish the object.

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/DescribeStmt.java [hunk 2]
```diff
@@ -63,6 +64,13 @@ public class DescribeStmt extends ShowStmt {
                     .addColumn(new Column("Table", ScalarType.createVarchar(30)))
                     .build();
 
+    private static final ShowResultSetMetaData DESC_TABLE_FUNCTION_TABLE_META_DATA =
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/DescribeStmt.java:67]: Why not split this class into multiple sub-classes?
DescOlapTableStmt, DescMVStmt, DescFilesTableFuncStmt, DescCredentialStmt, ...

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java [hunk 3]
```diff
@@ -331,6 +338,24 @@ public Void visitDescTableStmt(DescribeStmt node, ConnectContext context) {
...
+                table = new TableFunctionTable(node.getTableFunctionProperties());
+            } catch (DdlException e) {
+                throw new StorageAccessException(e);
+            }
+
+            List<Column> columns = table.getFullSchema();
+            for (Column column : columns) {
+                List<String> row = Arrays.asList(
+                        column.getName(),
+                        column.getType().canonicalName().toLowerCase(),
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java:353]: UPPER case?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java [hunk 4]
```diff
@@ -331,6 +338,24 @@ public Void visitDescTableStmt(DescribeStmt node, ConnectContext context) {
...
+                        column.isAllowNull() ? "YES" : "NO");
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java:354]: Use a constant variable? Should keep the string case.

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/DescribeStmt.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/DescribeStmt.java:104]: `DescribeStmt` is only for `describe table`.
`files()` is a table function table, so i put it here.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java (hunk 3 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java:353]: https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java#L447

In MySQL, the Type column should lowercase, and the Null column should uppercase.
keep consistent with olap table?

File: fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java [hunk 5]
```diff
@@ -100,6 +100,9 @@ public class FeConstants {
 
     public static final int DEFAULT_UNPARTITIONED_TABLE_BUCKET_NUM = 16;
 
+    public static final String YES = "YES";
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java:103]: These naming are too general. Maybe we can encapsulate them as:
public static class ShowSchemaConstants  {...}.

Or, will the YES and NO be used everywhere?

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java]: done

**Review by jaogoy — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wyb** [comment]
@Mergifyio rebase

**Review by stephen-shelby — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**